### PR TITLE
Add annotation access control at the folder level

### LINKED ISSUE #27: Need to be able to change all annotation access controls for a specific folder
If we want to change the permission controls for all annotations in a folder, say to add a new user or to make them public, there is no direct way to do this using existing API or UI. It would be nice (not sure if a high priority) to have some direct way to do this from the UI or API. Some version of the following would be nice to include directly into the UI/API.

```
DESIRED_PERMISSIONS = """{
  "groups": [
    {
      "description": "User Group for Mohamed's Annotation Project",
      "flags": [],
      "id": "????",
      "level": 0,
      "name": "brcaAnnotators"
    }
  ],
  "users": [
    {
      "flags": [],
      "id": "????",
      "level": 2,
      "login": "????",
      "name": "admin admin"
    }
}"""
for sldno, sld_girderID in enumerate(sldidxs):
    
    print("Slide %d of %d: %s" % (sldno, N_SLIDES, sld_girderID))

    # get annotations for this slide
    slideAnnotations = gc.get('/annotation?itemId={}'.format(sld_girderID))
    
    # go through annotations and edit permissions
    for ann in slideAnnotations:
        ann_girderID = ann['_id']
        _ = gc.put('/annotation/%s/access?access=%s' % 
                 (ann_girderID, DESIRED_PERMISSIONS))

```
where sldidxs is a list of girder IDs for all slides that belong to a folder and its subfolders (recursively).

### PR DESCRIPTION
**Pull request by eagw**
Adds the ability to change the access list for all annotations in a folder (and optionally all annotations in the subfolders of the folder).  The control can be found as an entry in the 'Folder actions' dropdown menu.

Closes #27

### CODE REVIEW

**Comment by manthey** [comment]
In retrospect, this really should be in the large_image repo in the girder_annotation module.

File: histomicsui/web_client/views/HierarchyWidget.js [hunk 1]
```diff
@@ -1,13 +1,153 @@
...
+                sortDir: -1,
+                recurse: false
+            }).done(() => {
+                root.collection.each((model) => {
+                    root.users.add({'_id': model.get('creatorId')});
+                });
+                $.when.apply($, root.users.map((model) => {
+                    return model.fetch();
+                })).always(() => {
+                    root.render();
```
Comment (manthey) [inline histomicsui/web_client/views/HierarchyWidget.js:80]: I'm seeing multiple render calls which makes the page flash quite a bit.  I'll run down what is going on.

**Comment by manthey** [comment]
> > In retrospect, this really should be in the large_image repo in the girder_annotation module.
> 
> Would you like me to move it there?

Yes, please.

**Comment by eagw** [comment]
With [#905](https://github.com/girder/large_image/pull/905) in large image, change is no longer needed here.

**Pull request closed without merging**